TICKET-6:

simyara opened the pull request:
related to ticket [Ticket-6](https://trello.com/c/gbpCw3Ym/4-06-add-remove-and-create-api-method)
pull request updated

simyara committed:
TICKET-6:
[+] add new picture
[+] delete picture

ikryvanos commented on actions/singlePicture.js:
```diff
@@ -69,5 +77,96 @@ module.exports = {
+            }
+        };
+
+        let validationResult = yield validator.validateObject(this.request.body, schema);
```
Why you need to validate such schema before main one?

ikryvanos commented on actions/singlePicture.js:
```diff
@@ -69,5 +77,96 @@ module.exports = {
+        let pictureData = yield pictureServer.findOne(this.params.id, this.request.body);
+
+        if (pictureData) {
+            this.status = 404;
```
404 status code not suitable for such case

ikryvanos commented on services/pictureServer.js:
```diff
@@ -88,7 +91,15 @@ module.exports = {
+    *deleteOne(id){
+        picItems = picItems.filter((item) => item.id !== id);
+        return id;
+        }
```
Wrong tabulation

ikryvanos commented on services/validator.js:
```diff
@@ -34,6 +34,36 @@ function validate(obj, schema) {
 }
 
+//all fields in schema marked as required must be presented in object
+function validateRequired(obj, schema) {
```
Make only one method for validation

simyara committed:
TICKET-6:
[*] transform 2 validation methods for 1
[*] error code when trying to add item with an existing id

simyara committed:
TICKET-6:
[*] tabulation fix

satamail commented on services/validator.js:
```diff
@@ -1,52 +1,38 @@
+                let res = validateObject(obj[key], schema[key].properties);
+                return res.errorMessage;  //(*) recursion
+            }
+            errorMessage = key + ' type mismatch';
```
Try to avoid error message duplication

simyara committed:
TICKET-6:
[*] order validation rules
